api/src/main/java/org/apache/iceberg/SnapshotUpdate.java

```diff
@@ -52,4 +53,13 @@
    */
   ThisT stageOnly();
 
+  /**
+   * Use a particular executor to access manifests. The default worker pool will be used by default.
+   *
+   * @param executorService the provided executor.
+   * @return this for method chaining.
+   */
+  default ThisT accessManifestsWith(ExecutorService executorService) {
```

Comment:
Could you separate this into a PR for the scan updates and a PR for some of the operation updates (like ExpireSnapshots)? This one together is really large.